ssr/src/page/wallet/txn.rs

```diff
@@ -137,101 +219,278 @@ pub mod provider {
                 canisters,
-                user_canister,
+                user_principal,
+                source,
             }
         }
     }
-    #[cfg(not(feature = "mock-wallet-history"))]
+    // #[cfg(not(feature = "mock-wallet-history"))]
     mod canister {
-        use super::{Canisters, CursoredDataProvider, TxnInfo, TxnTag};
+        use super::{
+            Canisters, CursoredDataProvider, IndexOrLedger, TokenBalance, TxnInfoType,
+            TxnInfoWallet,
+        };
         use crate::component::infinite_scroller::PageEntry;
-        use candid::Principal;
+        use candid::{Nat, Principal};
         use ic_agent::AgentError;
-        use yral_canisters_client::individual_user_template::{
-            HotOrNotOutcomePayoutEvent, MintEvent, Result15, TokenEvent,
+        use leptos::ServerFnError;
+        use yral_canisters_client::{
+            sns_index::{
+                Account, GetAccountTransactionsArgs, GetTransactionsResult, Transaction,
+                TransactionWithId,
+            },
+            sns_ledger::GetTransactionsRequest,
         };
 
-        fn event_to_txn(event: (u64, TokenEvent)) -> Option<TxnInfo> {
-            let (amount, tag) = match event.1 {
-                TokenEvent::Stake { amount, .. } => (amount, TxnTag::BetPlaced),
-                TokenEvent::Burn => return None,
-                TokenEvent::Mint {
-                    amount,
-                    details: MintEvent::NewUserSignup { .. },
-                    ..
-                } => (amount, TxnTag::SignupBonus),
-                TokenEvent::Mint {
-                    amount,
-                    details: MintEvent::Referral { .. },
-                    ..
-                } => (amount, TxnTag::Referral),
-                TokenEvent::Transfer { amount, .. } => (amount, TxnTag::Transfer),
-                TokenEvent::Receive { amount, .. } => (amount, TxnTag::HotorNotAccountTransfer),
-                TokenEvent::HotOrNotOutcomePayout {
-                    amount,
-                    details: HotOrNotOutcomePayoutEvent::CommissionFromHotOrNotBet { .. },
+        fn parse_transactions(
+            txn: TransactionWithId,
+            user_principal: Principal,
+        ) -> Result<TxnInfoWallet, ServerFnError> {
+            let timestamp = txn.transaction.timestamp;
+            let id = txn.id.0.to_u64_digits()[0];
+
+            match txn.transaction {
+                Transaction {
+                    mint: Some(mint), ..
+                } => {
+                    if is_user_principal(mint.to.owner) {
+                        Ok(TxnInfoWallet {
+                            tag: TxnInfoType::Mint { to: mint.to.owner },
+                            timestamp,
+                            amount: TokenBalance::new_cdao(mint.amount),
+                            id,
+                        })
+                    } else {
+                        Err(ServerFnError::new("Mint 'to' is not a user principal"))
```

Comment:
Do we really care? Principal ids are opaque and ideally you shouldn't need this lookup

Reply:
Yeah thats for filtering canister transactions out...

Reply:
Do we have a significant number of these?

Airdrop transfers will also show up as canister transactions (as its a transfer from the canister to the user principal), for example.

I dont see why we'd not want to show them unless they generate alot of spam

Finally, Internet Computer explicitly discourages to identify principals like this
https://docs.rs/ic-agent/latest/ic_agent/export/struct.Principal.html

Reply:
Yeah we do have a lot of these and because during minting we allocate extra tokens and mint extra tokens they do end up  being shown on the index and the ledger transactions and we dont wanna show it, so this is the only workaround we have currently...

Reply:
we can push this for now if its urgent but ideally we can discuss a few solutions in the product call tomorrow
cc: @harshita-srivastava-yral